.github/workflows/aws-terraform.yml

```diff
@@ -4,42 +4,37 @@ on:
   workflow_call:
     inputs:
       docker_image:
-        description: 'The runner image to be used'
+        description: "The runner image to be used"
         required: true
         type: string
       github_environment:
-        description: 'Defines the github environemnt for secrets'
+        description: "Defines the GitHub environment for secrets"
         required: true
         type: string
       deployment_environment:
-        description: 'Defines the deployment environemnt for makefile'
+        description: "Defines the deployment environment for Makefile"
         required: true
         type: string
       tf_action:
-        description: 'Terraform action to perform'
+        description: "Terraform action to perform"
         required: true
         type: string
       tf_backend_bucket_name:
-        description: 'Terraform backend bucket Name'
-        required: true
+        description: "Terraform backend bucket Name"
+        required: false
         type: string
       tf_backend_bucket_key:
-        description: 'Terraform backend bucket Key'
-        required: true
+        description: "Terraform backend bucket Key"
+        required: false
         type: string
       tf_backend_bucket_region:
-        description: 'Terraform backend bucket Region'
-        required: true
+        description: "Terraform backend bucket Region"
+        required: false
         type: string
       aws_region:
-        description: 'AWS region'
+        description: "AWS region"
```

Comment:
we can do some validation here. Only allowed valid AWS regions

Reply:
`workflow_call` does not support valid values list